File: mkdocs/overrides/devbook/start/hello-world.py
import json
import urllib.request

from symbolchain.facade.SymbolFacade import SymbolFacade
from symbolchain.symbol.Network import NetworkTimestamp

facade = SymbolFacade('mainnet')
print(f"Network name: {facade.network.name}")
# NetworkTimestamp(0) is the genesis block timestamp
launch_date = facade.network.to_datetime(NetworkTimestamp(0))
print(f"Network launch date: {launch_date}")

NODE_URL = 'https://001-sai-dual.symboltest.net:3001'
print(f'Using node {NODE_URL}')
try:
    # Fetch current chain information
    info_path = '/chain/info'
    print(f'Fetching chain information from {info_path}')
    with urllib.request.urlopen(
        f'{NODE_URL}{info_path}', timeout = 10
    ) as response:
        response_json = json.loads(response.read().decode())
        height = int(response_json['height'])
        print(f"  Blockchain height: {height:,} blocks")

except urllib.error.URLError as e:
    print(e.reason)

File: mkdocs/overrides/devbook/start/hello-world.mjs
import {
	SymbolFacade,
	NetworkTimestamp
} from 'symbol-sdk/symbol';
// [>step-1]
const facade = new SymbolFacade('mainnet');
console.log(`Network name: ${facade.network.name}`);
// NetworkTimestamp(0) is the genesis block timestamp (network launch)
const launchDate = facade.network.toDatetime(new NetworkTimestamp(0));
console.log(`Network launch date: ${launchDate.toISOString()}`); // [<step-1]
// [>step-2]
const NODE_URL = 'https://reference.symboltest.net:3001';
console.log(`Using node ${NODE_URL}`);
try {
	// Fetch current chain information
	const infoPath = '/chain/info';
	console.log(`Fetching chain information from ${infoPath}`);
	const response = await fetch(`${NODE_URL}${infoPath}`,
		{ timeout: 10000 });
	if (!response.ok) {
		throw new Error(`HTTP error! status: ${response.status}`);
	}
	const responseJson = await response.json();
	const height = parseInt(responseJson.height, 10);
	console.log(`  Blockchain height: ${height.toLocaleString()} blocks`);
} catch (e) {
	console.error(e.message, '| Cause:', e.cause?.code ?? 'unknown');
} // [<step-2]
